crates/harp/src/routines.rs

```diff
@@ -5,18 +5,32 @@
 //
 //
 
+use std::sync::Mutex;
+
 use libr::R_CallMethodDef;
 use libr::R_getEmbeddingDllInfo;
 use libr::R_registerRoutines;
 use log::error;
 
-static mut R_ROUTINES: Vec<R_CallMethodDef> = vec![];
+// Sync and Send wrapper that we can store in a global. Necessary because
+// `R_CallMethodDef` includes raw pointers.
+struct CallMethodDef {
+    inner: R_CallMethodDef,
+}
+
+unsafe impl Send for CallMethodDef {}
+unsafe impl Sync for CallMethodDef {}
```

Comment:
I think my main comment is that `#[harp::register]` is pretty freaking magical. It generates some code that runs _at application startup time_ in an OS specific way. 

I guess we can assume that code is always run on a single thread? I have no idea how startup code gets run on a per OS basis. I just want to make sure this doesn't screw us up.

Reply:
We're just saying it is fine to send these raw pointers across threads, which it is as long as you don't do anything with them. We're not introducing any new area of unsafety AFAICT.